lib/streamlit/DeltaGenerator.py

```diff
@@ -1876,6 +1881,39 @@ def slider(
             current_value = current_value[0] if single_value else current_value
         return current_value if single_value else tuple(current_value)
 
+    @_with_element
+    def file_uploader(self, element, label, type=['*'], key=None):
+
+        """Display a file uploader widget.
+
+        Parameters
+        ----------
+        label : str or None
+            A short label explaining to the user what this file uploader is for.
+        type : string[] or None
+            Array of allowed extensions. (['.png', '.jpg'])
+            Default value: ['*']
```

Comment:
What does `type=None` mean? Is that the same as `['*']`?

I think we shouldn't have `['*']` and the default should be `None` — which would mean that _all_ file types are allowed.

Reply:
Also, why are there periods in the name. Adds nothing, right?